src/gdb_main.c

```diff
@@ -423,35 +423,37 @@ static void exec_q_supported(const char *packet, const size_t length)
 
 static void exec_q_memory_map(const char *packet, const size_t length)
 {
+	target_s *target = cur_target;
+
 	(void)packet;
 	(void)length;
 	/* Read target XML memory map */
-	if ((!cur_target) && last_target) {
-		/* Attach to last target if detached. */
-		cur_target = target_attach(last_target, &gdb_controller);
+	if (!target) {
+		target = last_target;
 	}
-	if (!cur_target) {
+	if (!target) {
 		gdb_putpacketz("E01");
 		return;
 	}
 	char buf[1024];
-	target_mem_map(cur_target, buf, sizeof(buf)); /* Fixme: Check size!*/
+	target_mem_map(target, buf, sizeof(buf)); /* Fixme: Check size!*/
 	handle_q_string_reply(buf, packet);
 }
 
 static void exec_q_feature_read(const char *packet, const size_t length)
 {
+	target_s *target = cur_target;
```

Comment:
same comment as above about the ordering of parameter use suppression and variable declaration